x-pack/filebeat/input/s3/input_test.go

```diff
@@ -0,0 +1,246 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package s3
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"testing"
+
+	awssdk "github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+	"github.com/aws/aws-sdk-go-v2/service/s3/s3iface"
+	"github.com/aws/aws-sdk-go-v2/service/sqs"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/elastic/beats/libbeat/beat"
+)
+
+// MockS3Client struct is used for unit tests.
+type MockS3Client struct {
+	s3iface.ClientAPI
+}
+
+var (
+	s3LogString1 = "36c1f test-s3-ks [20/Jun/2019] 1.2.3.4 arn:aws:iam::1234:user/kaiyan.sheng@elastic.co 5141F REST.HEAD.OBJECT Screen1.png \n"
+	s3LogString2 = "28kdg test-s3-ks [20/Jun/2019] 1.2.3.4 arn:aws:iam::1234:user/kaiyan.sheng@elastic.co 5A070 REST.HEAD.OBJECT Screen2.png \n"
+	mockSvc      = &MockS3Client{}
+	info         = s3Info{
+		name:   "test-s3-ks",
+		key:    "log2019-06-21-16-16-54",
+		region: "us-west-1",
+	}
+	p = Input{}
+)
```

Comment:
I just noticed this. Do never define globals for the object under test. All tests must be isolated, so to ensure we don't have dependencies between test runs or internal state.

Reply:
Fixed! Thanks, I will remember that!